sorting/counting_sort_2.c

```diff
@@ -0,0 +1,33 @@
+#include<stdio.h>

```

Comment:
Please describe how the algorithm works, a Wikipedia link, what the algorithm does, etc..